### ISSUE TITLE
`ern` command tab completion

### ISSUE THREAD

**Issue by swashcap** [electrode-io/electrode-native#99]
Feature idea: there’s a lot of commands hanging off `ern`. It would be nice to have tab-completion for commands, which could help with use of the tool by reducing mistypes and limiting the command-not-found-so-show-help user errors.

**Comment by belemaire** [comment]
Yes we also discussed internally about this for a while now.

We need to add it at some point, first need to experiment with a POC.
The only "problem" we see is that it requires edits to the bash profile as it involves some scripting that needs to be handled at the shell level. Need to see if this is required only once or if it needs to be updated at some point (new commands or new version of yargs or what else). Need to decide as well if we want to tell the user to manually add the scripting to the profile or if its better too automate it. We'll look into it more when we have time.

**Comment by swashcap** [comment]
:+1: Sounds good. You _could_ maintain a `ern-completion.bash` script _inside_ `ern-local-cli` such that completion aligns with all available commands/subcommands of the existing installation. Users generally like to maintain their own bash profile (often in some sort of version control), and mutating it programmatically makes for a bad experience. I'd suggest opt-in. We could always have a “include completion” message on npm‘s `postinstall` hook so users know it's available:

```
Add the following lines to your ~/.bash_profile to include ern command tab-completion:

    if [ -f ~/.ern/completion.bash ]; then
      . ~/.ern/completion.bash
    fi

See more documentation here:
    https://electrode.io/native/cli/completion
```

Auto-complete inclusion in the docs and `--help` output would be nice, too.

**Comment by swashcap** [comment]
Looks like yargs [supports completion script dumping](https://github.com/yargs/yargs/blob/HEAD/docs/api.md#completioncmd-description-fn), which might make this easier. [Example use](https://stackoverflow.com/a/32260681).

**Comment by belemaire** [comment]
Moving to backlog, in an effort to reduce lingering open issues
